### ISSUE TITLE
Proposed topics/subtopics

### ISSUE THREAD

**Issue by kwstat** [friendly/statquotes#21]
I am preparing a file of quotes and would like to propose the following topics/subtopic:
```
Data, Data analysis
Data, Models
Data, Outliers
Statistics, Significance
Statistics, Bayesian
Statistics, Experimental design
Probability
```
What do you think?

**Comment by friendly** [comment]
That would be good, and I'd welcome you as a contributor and/or package author.

Some thoughts for further development:
* Current method of adding new quotes is kludgy. Requires to add all quotes to a single .csv file.
* Current system of topics & subtopics is OK, but better would be some system of using tags, not needing to be in a hierarchical structure.

Let's see how to add your quotes as a first step.

**Comment by kwstat** [comment]
1. I like the idea of tags for topics.

I explored some github repositories with similar quotations functionality in non-R languages:

https://github.com/JamesFT/Database-Quotes-JSON
https://github.com/NikhilNamal17/popular-movie-quotes
https://github.com/lukePeavey/quotable
https://github.com/vinitshahdeo/inspirational-quotes
https://github.com/scrapy/quotesbot
https://github.com/dwyl/quotes/blob/main/quotes.json

Most of the repos use a json file to store the quotes in a text file. The last one (on the above list) includes tags. The file looks like this:

[  {
    "author": "Albert Einstein",
    "source": "https://en.wikiquote.org/wiki/Albert_Einstein",
    "tags": "intelligence, knowledge, simplicity, understanding",
    "text": "If you can't explain it simply, you don't understand it well enough."
  },
  { next quote, etc } ]

Json has a LOT of markup (braces, quotes, colons), so it is easy to parse in many languages. But I think the format is overly complex for a simple quote file and it is tedious to edit by hand (e.g. escaped quote marks).

I think it would be simpler to use the current quote files and just add another line for tags.  Like this:

Statistical thinking will one day be as necessary for efficient citizenship as the ability to read and write.
--- H.G. Wells, Mankind in Making, 1903
%source: https://www.quora.com/What-is-the-source-of-the-H-G-Wells-quote-Statistical-thinking-will-one-day-be-as-necessary-for-efficient-citizenship-as-the-ability-to-read-and-write
tags: statistics, history

The readQuotes function could be modified to put tags in a new column of the dataframe and then searching tags could be something like `grepl("history", data$tags)`

2. If all the quotes were stored in txt files, then the readQuotes() function could loop over all of the .txt files in the inst directory and combine them into a single file which could be an Rdata file for the package.

I am happy to work on both of these (after spring break) if you are favorable.

**Comment by friendly** [comment]
Are you ready to do some work on this?

Looking over the `readQuotes` code, it would not be hard to introduce tags using a unique prefix string, but not `%` because that is treated as a comment.  How about:

    #tags: intelligence; knowledge; simplicity; understanding

To see how this works, perhaps you could markup your `quotes-kw.txt` file with tags?

For a longer source attribution, there is already the brief `source` field extracted from the `--- ` line.  Could do something like:

    #ref: [https://www.quora.com/What-is-the-source-of-the-H-G-Well ...

**Comment by kwstat** [comment]
I have removed the topic/subtopic sections and replaced them with more flexible tags.

Also:

- Combined all quote files into a single file: quotes_raw.txt.
- Added many quotes about hypothesis testing.
- Added 'tags' field to replace 'topic' and 'subtopic'.
- Added URLs for many quotes.
- Function `quote_topics` renamed to `quote_tags`.

There a lots of changes, so before I send you a pull request, I suggest you install the package from my repository and give it a try.  Let me know what feedback you have.
```
devtools::install_github("kwstat/statquotes")
```

**Comment by kwstat** [comment]
As I've thought about this a little more, I think I would rather remove the concept of "tags", and instead use "topic" as tags.  I.e. the quote file would still have a comma-separated line of topics (instead of calling them tags). This provides a little more backward compatibility.

**Comment by friendly** [comment]
OK; sorry -- I've been preoccupied with a paper on the history of EDA, now nearly done.
I'll look at your version of the package.

**Comment by friendly** [comment]
I've looked at your `kwstat/statquotes` repo and like it very much. The use of tags (`quo:`, `src:`, ..) for the `.txt` file is a lot more straight-forward for parsing and adding new quotes. We could even add a `suggest a quote` feature to allow others to add new quotes, probably in the form of an issue.

Looking at your code, it seems that you are actually using "tag", not "topic".  I'm fine with that. the main thing is that supersedes the hierarchy of topic / subtopic I had used before.

To proceed, I suggest:
* It would be easier (at least for me) to have your work as a new branch, say `dev-tags` under the `friendly/statquotes` repo, to which you will have access; that makes it easier to maintain in RStudio as versions of the same package.
* After some testing, then we can move `dev-tags` to `master`, and go from there.
* (However, there is a lot of non-overlap between our versions and I'm not sure exactly how to do the last step cleanly. Perhaps it is as simple as renaming `master` -> `master-old` and `dev-tags` -> `master`)
* Because this is a major change, bump the pkg version to 0.3.0.


Let me know if this is OK with you.

**Comment by friendly** [comment]
I have now pulled your `kwstat/statquotes` repo into `friendly/statquotes` as the `dev-tags` branch.  Bumped the package version and made a few other small changes (see NEWS)

**Comment by kwstat** [comment]
I have spent a lot of time on the `dev-tags` branch and I believe it is ready to go mainstream.  I reviewed the previous changes to the `master` branch and found the addition of the `table` argument to `quote_tags()`, so I have incorporated that into the `dev-tags` branch as well. Some of the changes that I have made are to follow "best practices" with packages: using `data-raw`, cleaning out `inst`, using `NEWS.md` and adding `CONTRIBUTING.md`.

I think that it should be possible to rename `master` to `master-old` and `dev-tags` to `master`.
